### PR TITLE
mknum.pl: Exclude duplicate entries and include source file names in diagnostics

### PR DESCRIPTION
**Pull request by DDvO**
From the C language perspective it is not a problem if a function is declared multiple times identically (be it in the same or different source/header files).
Yet usually it is a mistake to do so.

If this happens for an exported library symbol, `mknum.pl` (called by `make ordinals`) excludes more than two occurrences, but in case of two occurrences so far produces an entry for _both_ of them, which will cause problems.

This PR adds a check for identical entries and makes `mknum.pl` stop with an error if found.
It also adds diagnostic information in which source files multiple declarations were found.

### CODE REVIEW

File: util/perl/OpenSSL/Ordinals.pm [hunk 1]
```diff
@@ -485,12 +497,12 @@ sub add {
...
     # For the caller to show
     my @returns = ( $new_item );
-    push @returns, $self->add_alias($alias->{name}, $name, @{$alias->{defs}})
+    push @returns, $self->add_alias($source, $alias->{name}, $name, @{$alias->{defs}})
         if defined $alias;
     return @returns;
 }
 
-=item B<< $ordinals->add_alias ALIAS, NAME, LIST >>
+=item B<< $ordinals->add_alias SOURCE, ALIAS, NAME, LIST >>
```
Comment (levitte) [inline util/perl/OpenSSL/Ordinals.pm:505]: Please document SOURCE

File: util/perl/OpenSSL/Ordinals.pm [hunk 2]
```diff
@@ -734,7 +749,7 @@ Available options are:
 
 =over 4
 
-=item B<< from => STRING >>
+=item B<< source => FILENAME >>, B<< from => STRING >>
```
Comment (levitte) [inline util/perl/OpenSSL/Ordinals.pm:752]: Please document FILENAME

File: util/perl/OpenSSL/Ordinals.pm [hunk 3]
```diff
@@ -757,8 +772,8 @@ STRING must conform to the following EBNF description:
 
 (C<letter> and C<digit> are assumed self evident)
 
-=item B<< name => STRING >>, B<< number => NUMBER >>, B<< version => STRING >>,
-      B<< exists => BOOLEAN >>, B<< type => STRING >>,
+=item B<< source => STRING >>, B<< name => STRING >>, B<< number => NUMBER >>,
```
Comment (levitte) [inline util/perl/OpenSSL/Ordinals.pm:775]: For consistency, shouldn't that be `B<< source => FILENAME >>` ?

File: util/perl/OpenSSL/Ordinals.pm (hunk 1 above)
Comment (DDvO) [reply util/perl/OpenSSL/Ordinals.pm:505]: Done

File: util/perl/OpenSSL/Ordinals.pm (hunk 2 above)
Comment (DDvO) [reply util/perl/OpenSSL/Ordinals.pm:752]: Done

File: util/perl/OpenSSL/Ordinals.pm (hunk 3 above)
Comment (DDvO) [reply util/perl/OpenSSL/Ordinals.pm]: Changed

**Review by levitte — APPROVED** [review]

**Comment by DDvO** [comment]
Merged

**Pull request closed without merging**